specification/sql/resource-manager/Microsoft.Sql/preview/2022-02-01-preview/ManagedInstanceDtcs.json

```diff
@@ -0,0 +1,369 @@
+      },
+      "put": {
+        "tags": [
+          "ManagedInstanceDtcs"
+        ],
+        "description": "Updates the managed instance DTC settings.",
+        "operationId": "ManagedInstanceDtcs_CreateOrUpdate",
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ResourceGroupParameter"
+          },
+          {
+            "$ref": "#/parameters/ManagedInstanceNameParameter"
+          },
+          {
+            "name": "dtcName",
+            "in": "path",
+            "description": "The name of the managed instance DTC.",
+            "required": true,
+            "type": "string",
+            "enum": [
+              "current"
+            ],
+            "x-ms-enum": {
+              "name": "DtcName",
+              "modelAsString": true
+            }
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "description": "Managed instance DTC settings.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ManagedInstanceDtc"
+            }
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully updated DTC settings.",
+            "schema": {
+              "$ref": "#/definitions/ManagedInstanceDtc"
+            }
+          },
+          "default": {
+            "description": "*** Error Responses: ***\n\n * 400 InvalidDtcName - User tried to interact with Azure SQL Managed Instance DTC settings with a DTC name different from 'current'.\n\n * 400 InvalidUpdateDtcOperationParameters - The provided set of parameters in the body of the request is either empty or invalid.\n\n * 400 InvalidDtcAuthenticationMethod - DTC authentication method \"{0}\" is not supported.\n\n * 400 ManagementServiceManagedInstanceDtcFeatureDisabled - User tried to interact with Azure SQL Managed Instance DTC settings but DTC feature is currently not supported.\n\n * 400 InvalidDtcVirtualClusterVersion - Virtual cluster version of the Azure SQL Managed Instance \"{0}\" is not supported.\n\n * 404 ResourceNotFound - The requested resource was not found.\n\n * 404 OperationIdNotFound - The operation with Id does not exist.\n\n * 404 OperationIdNotFound - The operation with Id does not exist.\n\n * 409 OperationCancelled - The operation has been cancelled by user.\n\n * 409 OperationInterrupted - The operation on the resource could not be completed because it was interrupted by another operation on the same resource.\n\n * 500 OperationTimedOut - The operation timed out and automatically rolled back. Please retry the operation."
+          },
+          "202": {
+            "description": "Applying DTC settings is in progress"
+          }
```

Comment:
Recommended pattern for put async is 200/201 + provisioning state, 202 is old model. See [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#creatingupdating-using-put](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#creatingupdating-using-put)

Reply:
Since 201 is the "Created" code, I thought that it might be misleading to use provisioning state, since the DTC resource in question cannot be created by the API users. That resource will be created alongside managed instance, and Its settings can only be updated.

Reply:
You can just use 200 + provisioning. Please see the above link: "Since provisioning is not complete, the PUT response body MUST contain a provisioningState property set to a non-terminal value (as described below). Resource properties should reflect the update that is in progress (i.e. the state the resource will be in once the async operation is complete)."

Reply:
I've took another look at the code, and I see that changing this would require changing some internal workflows code which I do not own (since the controllers read the provisioningState value from an internal workflow object). Could we keep the 202 model?

Reply:
Its a good practice to first get your models reviewed before you code up your solution so you have time to address the feedback provided here. We are moving towards a direction where we are able to give our customers consistent control plane experience across all services. Having these one-off implementations which are different leads to bad customer experience and should be avoided, Please check what it takes to make the necessary changes on the service to accommodate this ask. If u cant accommodate this, pl let provide a business justification on why it cannot be implemented.

Reply:
Also ARM template deployments that issue a PUT on the resource for an update, will look for the terminal state of the operation using the provisioningState property. So you will lose the ARM template functionality if you dont implement provisioningState

Reply:
Hmm we didn't know that the ARM template functionality would be lost if not implementing provisioning state. I've added the provisioning state.